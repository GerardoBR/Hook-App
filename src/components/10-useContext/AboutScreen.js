import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {
    const {user, setsUser} = useContext(UserContext);
    const handlClik =()=>{
        setsUser({
            
        })
    }
    return (
        <div>
            <h1> About  Screen</h1>
            <hr/>
            <pre>
                {
                    JSON.stringify( user, null, 3 )
                }
            </pre>
            <button className = "btn btn-warning" onClick={ handlClik }> Logout</button>
        </div>
    )
}
