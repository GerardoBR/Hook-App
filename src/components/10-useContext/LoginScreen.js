import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
    // leer context 
    //  ocupar el setUser 
    const {setsUser, user} = useContext(UserContext)
    return (
        <div>
            <h1> Login Screen</h1>
            <hr/>
            <button className="btn btn-primary" onClick= {()=> setsUser({
                id: 123,
                name : 'Cesar'
            })}>Login</button>
        </div>
    )
}
