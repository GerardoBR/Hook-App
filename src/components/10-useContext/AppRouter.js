import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <>
                <NavBar/>
                <div className= "container">

                    <Switch>
                        <Route
                            exact
                            path = "/"
                            component = { HomeScreen }
                        >
                        </Route>
                        <Route
                            path = "/about"
                            component = { AboutScreen }
                        >
                        </Route>
                        <Route
                            path = "/login"
                            component = { LoginScreen }
                        >
                        </Route>
                        {/* <Route component = { HomeScreen}/> */}
                        <Redirect to ="/"/>
                    </Switch>
                </div>

            </>
        </Router>
    )
}
