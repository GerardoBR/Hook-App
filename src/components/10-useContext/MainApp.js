import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainApp = () => {
    const [user, setsUser] = useState({});

    return (
        <UserContext.Provider value = { {
            user,
            setsUser
        } }>
            <AppRouter/>
        </UserContext.Provider>
    )
}
