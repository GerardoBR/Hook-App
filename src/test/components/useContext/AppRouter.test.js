import { mount } from "enzyme"
import { AppRouter } from "../../../components/10-useContext/AppRouter"
import { UserContext } from "../../../components/10-useContext/UserContext";

describe('Pruebas en AppRouter/>', () => {
    const user = {
        id : 1,
        name : 'Gerardo'
    }
    const wrapper = mount(
        <UserContext.Provider value = {{
            user
        }}>

            <AppRouter/>
        </UserContext.Provider>
    );
    test('debe de mostrarse coerrectamente ', () => {
        expect ( wrapper).toMatchSnapshot();
    })
    
})
