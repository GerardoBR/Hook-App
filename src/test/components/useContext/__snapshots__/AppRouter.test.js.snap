// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter/> debe de mostrarse coerrectamente  1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand selected"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        UseContext
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-item nav-link  selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link "
            href="/about"
            onClick={[Function]}
          >
             Abut
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link "
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
         Home Screen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Gerardo"
}
      </pre>
    </div>
  </div>,
]
`;
